packages/sdk-auth/src/tokenProvider.js

```diff
@@ -64,11 +69,29 @@ export default class TokenProvider {
         new Error('Method "fetchTokenInfo" was not provided')
       )
 
-    return Promise.resolve(this.fetchTokenInfo(this.sdkAuth))
+    // do not run fetchTokenInfo more than once at any given time and cache the result
+    if (this.fetchTokenInfoPromise) return this.fetchTokenInfoPromise
+
+    this.fetchTokenInfoPromise = Promise.resolve(
+      this.fetchTokenInfo(this.sdkAuth)
```

Comment:
woah, what is happening here, is it a nested promise?

Reply:
Hi @daern91  - the idea behind this code is, that `this.fetchTokenInfo()` can return a non-promise value so that's why is it wrapped into a promise.

Also, we use `this.fetchTokenInfoPromise` variable to lock this function so it won't fetch multiple tokens from API. Instead, it will start fetching token only on the first call and for all next calls, it will return the same promise so one result will be replicated to all function calls